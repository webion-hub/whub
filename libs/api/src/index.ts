import ContactUsEndpoint from "./lib/endpoints/ContactUsEndpoint";
import CustomersEndpoint from "./lib/CustomersEndpoint";

export const api = {
  contactUs: ContactUsEndpoint,
  customers: CustomersEndpoint,
}