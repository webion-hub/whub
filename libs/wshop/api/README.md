# wshop-api

This library was generated with [Nx](https://nx.dev).

## Building

Run `nx build wshop-api` to build the library.

## Running unit tests

Run `nx test wshop-api` to execute the unit tests via [Jest](https://jestjs.io).
