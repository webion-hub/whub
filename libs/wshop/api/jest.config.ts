/* eslint-disable */
export default {
  displayName: 'wshop-api',
  preset: '../../../jest.preset.js',
  globals: {
    'ts-jest': {
      tsconfig: '<rootDir>/tsconfig.spec.json',
    },
  },
  transform: {
    '^.+\\.[tj]s$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'js', 'html'],
  coverageDirectory: '../../../coverage/libs/wshop/api',
};
