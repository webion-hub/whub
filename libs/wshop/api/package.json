{
  "name": "@whub/wshop-api",
  "version": "0.0.1",
  "type": "commonjs"
}
