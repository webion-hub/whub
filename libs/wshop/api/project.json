{
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/wshop/api/src",
  "projectType": "library",
  "targets": {
    "build": {
      "executor": "@nrwl/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/wshop/api",
        "main": "libs/wshop/api/src/index.ts",
        "tsConfig": "libs/wshop/api/tsconfig.lib.json",
        "assets": ["libs/wshop/api/*.md"]
      }
    },
    "publish": {
      "executor": "@nrwl/workspace:run-commands",
      "options": {
        "command": "node tools/scripts/publish.mjs wshop-api {args.ver} {args.tag}"
      },
      "dependsOn": [
        {
          "projects": "self",
          "target": "build"
        }
      ]
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/wshop/api/**/*.ts"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/wshop/api"],
      "options": {
        "jestConfig": "libs/wshop/api/jest.config.ts",
        "passWithNoTests": true
      }
    }
  },
  "tags": []
}
