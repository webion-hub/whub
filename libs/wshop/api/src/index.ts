export { WShopApi } from './lib/WShopApi'
export { ApiConfig } from './lib/settings/api-config'
