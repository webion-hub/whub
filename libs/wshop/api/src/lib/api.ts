import axios, { AxiosInstance } from "axios";
import { ProductsEndpoint } from "./endpoints/ProductsEndpoint";
import { ApiConfig } from "./settings/api-config";

export class WShopApi {
  private readonly client: AxiosInstance;

  constructor(config: ApiConfig) {
    this.client = axios.create({
      baseURL: config.baseUrl,
    });
  }

  get Products() {
    return new ProductsEndpoint(this.client);
  }
}