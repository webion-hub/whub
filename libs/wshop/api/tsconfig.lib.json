{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "outDir": "../../../dist/out-tsc",
    "declaration": true,
    "types": []
  },
  "include": ["**/*.ts", "../../apis/core/src/lib/abstractions/ResponseMapper.ts"],
  "exclude": ["jest.config.ts", "**/*.spec.ts", "**/*.test.ts"]
}
