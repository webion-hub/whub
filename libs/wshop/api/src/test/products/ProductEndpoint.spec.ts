import { faker } from "@faker-js/faker";
import { api } from "../api"

const createProduct = async () => {
  const created = await api.products.create({
    name: faker.commerce.product(),
    code: faker.random.alphaNumeric(5),
  });

  return api.products.withId(created.data.id);
}


describe('Product endpoint', () => {
  faker.seed(101010);

  it('Should get a product', async () => {
    const product = await createProduct();
    const loaded = await product.load();
    
    expect(loaded).toMatchSnapshot();
  });

  it('Should update a product', async () => {
    const product = await createProduct();
    const updated = await product.update({
      name: 'Paolo',
      code: 'ABC123',
    });

    expect(updated).toMatchSnapshot();
  });

  it('Should delete a product', async () => {
    const product = await createProduct();
    const deleted = await product.delete();

    expect(deleted).toMatchSnapshot();
  });
});