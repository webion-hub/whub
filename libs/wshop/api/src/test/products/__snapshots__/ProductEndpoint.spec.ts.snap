// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product endpoint Should delete a product 1`] = `
Object {
  "config": Object {
    "adapter": [Function],
    "baseURL": "http://localhost:5003",
    "data": undefined,
    "env": Object {
      "FormData": [Function],
    },
    "headers": Object {
      "Accept": "application/json, text/plain, */*",
    },
    "maxBodyLength": -1,
    "maxContentLength": -1,
    "method": "delete",
    "timeout": 0,
    "transformRequest": Array [
      [Function],
    ],
    "transformResponse": Array [
      [Function],
    ],
    "transitional": Object {
      "clarifyTimeoutError": false,
      "forcedJSONParsing": true,
      "silentJSONParsing": true,
    },
    "url": "shop/products/9",
    "validateStatus": [Function],
    "xsrfCookieName": "XSRF-TOKEN",
    "xsrfHeaderName": "X-XSRF-TOKEN",
  },
  "data": "",
  "headers": Object {
    "content-length": "0",
  },
  "request": XMLHttpRequest {},
  "status": 200,
  "statusText": "OK",
}
`;

exports[`Product endpoint Should get a product 1`] = `
Object {
  "config": Object {
    "adapter": [Function],
    "baseURL": "http://localhost:5003",
    "data": undefined,
    "env": Object {
      "FormData": [Function],
    },
    "headers": Object {
      "Accept": "application/json, text/plain, */*",
    },
    "maxBodyLength": -1,
    "maxContentLength": -1,
    "method": "get",
    "timeout": 0,
    "transformRequest": Array [
      [Function],
    ],
    "transformResponse": Array [
      [Function],
    ],
    "transitional": Object {
      "clarifyTimeoutError": false,
      "forcedJSONParsing": true,
      "silentJSONParsing": true,
    },
    "url": "shop/products/1",
    "validateStatus": [Function],
    "xsrfCookieName": "XSRF-TOKEN",
    "xsrfHeaderName": "X-XSRF-TOKEN",
  },
  "data": Object {
    "attachments": Array [],
    "category": null,
    "code": "5zxqb",
    "createdAt": "0001-01-01T00:00:00",
    "description": null,
    "details": Array [],
    "id": 1,
    "images": Array [],
    "lastUpdatedAt": "0001-01-01T00:00:00",
    "mainVariant": null,
    "name": "Pants",
    "price": null,
    "producer": null,
    "rating": null,
    "relatedProducts": Array [],
    "tags": Array [],
    "variants": Array [],
  },
  "headers": Object {
    "content-type": "application/json; charset=utf-8",
  },
  "request": XMLHttpRequest {},
  "status": 200,
  "statusText": "OK",
}
`;

exports[`Product endpoint Should update a product 1`] = `
Object {
  "config": Object {
    "adapter": [Function],
    "baseURL": "http://localhost:5003",
    "data": "{\\"name\\":\\"Paolo\\",\\"code\\":\\"ABC123\\"}",
    "env": Object {
      "FormData": [Function],
    },
    "headers": Object {
      "Accept": "application/json, text/plain, */*",
      "Content-Type": "application/json",
    },
    "maxBodyLength": -1,
    "maxContentLength": -1,
    "method": "put",
    "timeout": 0,
    "transformRequest": Array [
      [Function],
    ],
    "transformResponse": Array [
      [Function],
    ],
    "transitional": Object {
      "clarifyTimeoutError": false,
      "forcedJSONParsing": true,
      "silentJSONParsing": true,
    },
    "url": "shop/products/8",
    "validateStatus": [Function],
    "xsrfCookieName": "XSRF-TOKEN",
    "xsrfHeaderName": "X-XSRF-TOKEN",
  },
  "data": "",
  "headers": Object {
    "content-length": "0",
  },
  "request": XMLHttpRequest {},
  "status": 200,
  "statusText": "OK",
}
`;
